src/chericat.c

```diff
@@ -30,174 +30,213 @@
+            "[[-l|--lib -v|--overview] | [-c|--compart -v|--overview]]\n\t"
+            "[[-l|--lib -i|--caps_info <library name>] | [-c|--compart -i|--caps_info <compartment name>]]\n"
+            "    database name    - name of the database to store data captured by chericat\n"
+            "    pid              - pid of the target process\n"
+            "    library name     - name of the library for which show the capabilities info\n"
+            "    compartment name - name of the compartment for which show the capabilities info\n"
+            "Options:\n"
+            "    -d Enable debugging output. Repeated -d's (up to 3) increase verbosity.\n"
+            "    -f Provide the database name to capture the data collected.\n"
+            "       If omitted an in-memory db is used\n"
+            "    -p Scan the vm blocks and persist the data to the provided database.\n"
+            "    -l Show the capabilities data in library-centric view.\n"
+            "    -c Show the capabilities data in compartment-centric view.\n"
+            "    -v Show the vm info, arranged in either library- or compartment-centric view\n"
+            "    -i Show capabalities found in the provided library or compartment\n");
+    exit(1);
 }
 
-static struct option long_options[] = 
+static struct option long_options[] =
 {
-	{"debug_level", no_argument, 0, 'd'},
-	{"database_name", required_argument, 0, 'f'},
-	{"scan_mem", required_argument, 0, 'p'},
-	{"caps_summary", no_argument, 0, 'v'},
-	{"rtld_linkmap_scan", required_argument, 0, 'l'},
-	{"caps_symbols_summary", required_argument, 0, 'c'},
-	{0,0,0,0}
+    {"debug", no_argument, 0, 'd'},
+    {"db", required_argument, 0, 'f'},
+    {"attach", required_argument, 0, 'p'},
+    {"lib", no_argument, 0, 'l'},
+    {"compart", no_argument, 0, 'c'},
+    {"overview", no_argument, 0, 'v'},
+    {"caps_info", required_argument, 0, 'i'},
+    {0,0,0,0}
 };
 
 void terminate_chericat(int sig)
 {
-	if (pid != -1) {
-		ptrace_detach(pid);
-	}
-
-	xo_finish();
-	
-	if (db != NULL) {
-		sqlite3_close(db);
-	}
-	exit(sig);
+    xo_finish();
+    if (db != NULL) {
+	sqlite3_close(db);
+    }
+    exit(sig);
 }
 
-int 
-main(int argc, char *argv[])
+int main(int argc, char **argv)
 {
-	// libxo API to parse the libxo command line arguments. They are removed once parsed and stored, 
-	// the program arguments would then be handled as intended without the libxo arguments.
-	argc = xo_parse_args(argc, argv);
-
-	int optindex;
-	int opt = getopt_long(argc, argv, "df:p:vl:c:", long_options, &optindex);
-
-	if (opt == -1) {
-		usage();
-		return 0;
+    // libxo API to parse the libxo command line arguments. They are removed once parsed and stored,
+    // the program arguments would then be handled as intended without the libxo arguments.
+    argc = xo_parse_args(argc, argv);
+   
+    int opt_f=0;
+    int opt_p=0;
+    int opt_l=0;
+    int opt_c=0;
+    int opt_v=0;
+    int opt_i=0;
```

Comment:
This is fine as is, but BSD utility style would make these globals and use default (.bss) initialization to zero them.

Regardless I'd suggest using `bool` unless they are levels.